### ISSUE TITLE
Issue with community posting in Movim

### ISSUE THREAD

**Issue by petrk94** [movim/movim#756]
Hello,
I have deployed Movim with docker and wanted try the community/groups function to subscribe it.
The Subscription of the new group work, but I can not create or post content as owner, even not if its open for all to post.

What can cause this problem?
Im using openfire as XMPP Server.

I hope you can help

**Comment by edhelas** [comment]
Possibly OpenFire. You can start the daemon in debug using -d -v as parameters. It will output the XMPP logs, there you can see in details what is happening when you try to create something on your Pubsub nodes.

**Comment by petrk94** [comment]
Hello,
I have it running over docker, so I could only get the debug via the OpenFire Admin Interface.
After enabling. I have reloaded Movim in the browser and got this information:
```
2018.12.26 00:02:30 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_RECEIVED to session 111 
Queue : [MESSAGE_RECEIVED, ] 

2018.12.26 00:02:30 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_RECEIVED to session 111 
Queue : [MESSAGE_RECEIVED, , MESSAGE_RECEIVED, ] 

2018.12.26 00:02:30 org.apache.mina.filter.codec.ProtocolCodecFilter - Processing a MESSAGE_RECEIVED for session 111 
2018.12.26 00:02:30 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_SENT to session 111 
Queue : [MESSAGE_RECEIVED, , MESSAGE_SENT, ] 

2018.12.26 00:02:30 org.apache.mina.filter.codec.ProtocolCodecFilter - Processing a MESSAGE_RECEIVED for session 111 
2018.12.26 00:02:30 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_SENT to session 111 
Queue : [MESSAGE_SENT, ] 

2018.12.26 00:02:30 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_SENT to session 111 
Queue : [MESSAGE_SENT, , MESSAGE_SENT, ] 

2018.12.26 00:02:30 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_SENT to session 111 
Queue : [MESSAGE_SENT, , MESSAGE_SENT, , MESSAGE_SENT, ] 

2018.12.26 00:02:31 org.quartz.core.QuartzSchedulerThread - batch acquisition of 0 triggers 

```

After clicking on creating a new post in the community group I created in movim before, I got the error that I can not post and the debug log showed this information:
```

2018.12.26 00:02:41 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_RECEIVED to session 111 
Queue : [MESSAGE_RECEIVED, ] 

2018.12.26 00:02:41 org.apache.mina.filter.codec.ProtocolCodecFilter - Processing a MESSAGE_RECEIVED for session 111 
2018.12.26 00:02:41 org.apache.mina.filter.executor.OrderedThreadPoolExecutor - Adding event MESSAGE_SENT to session 111 
Queue : [MESSAGE_SENT, ] 
```

I dont really understand what it means, but OpenFire is the only XMPP server which has a good admin interface and work on my server, others cause problems. So I hope it will work with Movim too.
Im running right now Openfire 4.2.3

I saw that there is now 4.3.0 as beta version (here is the changelog: http://download.igniterealtime.org/openfire/docs/latest/changelog.html)

After searching for XEP I found this information:
[OF-1579] - Add support for XEP-0368: SRV records for XMPP over TLS
[OF-827] - Add support for XEP-0191: Blocking Command
[OF-1472] - XEP-0157: Expose admin contacts

And this one, I suppose this is the reason why it fails?
[OF-1511] - Add Pubsub Publishing Options

**Comment by edhelas** [comment]
This doesn't gives me lots of information unfortunately. Is it possible to have a test account created on your server then I can reproduce?

**Comment by petrk94** [comment]
@edhelas 
you can test it here: http://207.180.229.137/?login
You can create a new account by yourself in a XMPP client, I tried it with Conversations, what works well.
As you can see on the login page of the movim instance, this is the server domain for the signup.
Its not bound to a domain for now.

**Comment by edhelas** [comment]
Publishing a post on your XMPP server works for my Movim instance.
Yours is crashing when I do so, please check the logs to see why this crash happens (error.log or syslog).

Also OpenFire is, for now, not an officially supported server for Movim, we still have a few issues with it (see https://issues.igniterealtime.org/projects/OF/issues/OF-1433?filter=allopenissues).

**Comment by edhelas** [comment]
```xml
<iq xmlns="jabber:client" from="edhelas@vmd31906.contaboserver.net/movimL1m9xT" to="pubsub.vmd31906.contaboserver.net" type="set" id="MXHmsT">
<pubsub xmlns="http://jabber.org/protocol/pubsub" xmlns:default="http://www.w3.org/2005/Atom">    
  <publish node="test">
    <item id="test_post">
      <default:entry xmlns="http://www.w3.org/2005/Atom"/>
    </item>
  </publish>
</pubsub>
</iq> : sent to XMPP
<iq type="error" id="MXHmsT" from="pubsub.vmd31906.contaboserver.net" to="edhelas@vmd31906.contaboserver.net/movimL1m9xT">
<pubsub xmlns="http://jabber.org/protocol/pubsub" xmlns:default="http://www.w3.org/2005/Atom">
  <publish node="test">
    <item id="test_post">
      <default:entry xmlns="http://www.w3.org/2005/Atom">
      </default:entry>
    </item>
  </publish>
</pubsub>
<error code="400" type="modify">
  <bad-request xmlns="urn:ietf:params:xml:ns:xmpp-stanzas"/>
    <item-forbidden xmlns="http://jabber.org/protocol/pubsub#errors"/>
  </error>
</iq> : received
```

Here is another bug. I'm the owner of the node but cannot publish an item to it. This needs to be investigated.

**Comment by guusdk** [comment]
I've started digging into this issue. I think that most issues mentioned here have been resolved in the beta release of Openfire 4.5.0 (maybe earlier, but the 4.5.0 Beta is what I'm testing with).

I noticed that when creating a new community and posting to it, posts still didn't show up (much as the original author reported). However, Openfire-sided, the posts were created as expected. I'm also not detecting any errors in the stanza logs.

Turns out that Openfire (or at least this instance) does not, by default, include the payload of pubsub items, when all items are queried. Instead, a list of items without payload is returned. This behavior can be configured in Openfire by toggling a setting for the corresponding pubsub node in the Openfire admin console (see screenshot)

![image](https://user-images.githubusercontent.com/4253898/71698206-f6712a80-2dba-11ea-8773-f1e67b9b9057.png)

With this, posts show up as expected!

I have yet to determine if the setting got applied in error by Openfire, or if Movim simply doesn't expect this to happen, or anything else. But, regardless: progress!

**Comment by guusdk** [comment]
For what it's worth, it appears that Movim explicitly configures the node to not include payloads (which might be simply an echo of a default form it got sent). See the `pubsub#deliver_payloads` field in this stanza:

```xml
<iq from="guus-movim@igniterealtime.org/movimCVq0Ih" id="fbltAU" to="pubsub.igniterealtime.org" type="set">
  <pubsub xmlns="http://jabber.org/protocol/pubsub">
    <create node="urn:xmpp:microblog:0:comments/a9ce0cfc-68b0-430b-9273-92db9fbda54c"/>
    <configure>
      <x type="submit" xmlns="jabber:x:data">
        <field type="hidden" var="FORM_TYPE">
          <value>http://jabber.org/protocol/pubsub#node_config</value>
        </field>
        <field var="pubsub#deliver_payloads">
          <value>false</value>
        </field>
        <field var="pubsub#persist_items">
          <value>true</value>
        </field>
        <field var="pubsub#access_model">
          <value>open</value>
        </field>
        <field var="pubsub#notify_retract">
          <value>true</value>
        </field>
        <field var="pubsub#max_items">
          <value>1000</value>
        </field>
        <field var="pubsub#send_last_published_item">
          <value>on_sub</value>
        </field>
        <field var="pubsub#publish_model">
          <value>open</value>
        </field>
      </x>
    </configure>
  </pubsub>
</iq>
```

**Comment by guusdk** [comment]
Hmm, that's a different node. Didn't capture the create of the original one. I'm not sure if the stanza above is representative (but there's a good chance that it is).

**Comment by guusdk** [comment]
When Movim configures a node to not include payloads, that configuration should only apply to event notifications, not to item retrieval. Openfire does not apply this configuration correctly. I have raised https://igniterealtime.atlassian.net/browse/OF-2543 to track this problem. A fix is in the works: https://github.com/igniterealtime/Openfire/pull/2147 and will likely be included in Openfire 4.7.5 / 4.8.0.

**Comment by edhelas** [comment]
Thanks ;)
